en/docs/develop/writing-a-custom-federated-authenticator.md

```diff
@@ -106,1262 +111,118 @@ public interface ApplicationAuthenticator extends Serializable {
-        public static final String FB_AUTHZ_URL = "AuthnEndpoint";
-        public static final String FB_TOKEN_URL = "AuthTokenEndpoint";
-        public static final String FB_USER_INFO_URL = "UserInfoEndpoint";
-        private FacebookCustomAuthenticatorConstants() {}
-        }
-        /*
-        * Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
-        *
-        * WSO2 Inc. licenses this file to you under the Apache License,
-        * Version 2.0 (the "License"); you may not use this file except
-        * in compliance with the License.
-        * You may obtain a copy of the License at
-        *
-        * http://www.apache.org/licenses/LICENSE­2.0
-        *
-        * Unless required by applicable law or agreed to in writing,
-        * software distributed under the License is distributed on an
-        * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-        * KIND, either express or implied. See the License for the
-        * specific language governing permissions and limitations
-        * under the License.
-        */
-        package org.wso2.carbon.identity.application.authenticator.social.internal;
-        import org.apache.commons.logging.Log;
-        import org.apache.commons.logging.LogFactory;
-        import org.osgi.service.component.ComponentContext;
-        import org.wso2.carbon.identity.application.authentication.framework.ApplicationAuthenticator;
-        import org.wso2.carbon.identity.application.authenticator.social.facebook2.FacebookCustomAuthenticator;
-        import java.util.Hashtable;
-        /**
-        * @scr.component name="identity.application.authenticator.facebook.component"
-        * immediate="true"
-        */
-        public class FacebookCustomAuthenticatorServiceComponent {
-        private static final Log LOGGER = LogFactory.getLog(FacebookCustomAuthenticatorServiceComponent.class);
-        protected void activate(ComponentContext ctxt) {
-        try {
-        FacebookCustomAuthenticator facebookAuthenticator = new FacebookCustomAuthenticator();
-        Hashtable props = new Hashtable();
-        ctxt.getBundleContext().registerService(ApplicationAuthenticator.class.getName(),
-            facebookAuthenticator, props);
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Facebook Custome Authenticator bundle is activated");
-        }
-        } catch (Throwable e) {
-        LOGGER.fatal(" Error while activating Facebook authenticator ", e);
-        }
-        }
-        protected void deactivate(ComponentContext ctxt) {
-        if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug("Facebook Custom Authenticator bundle is deactivated");
-        }
-        }
-        }
-        ```
+1. Start the server and log in to the WSO2 IS Management Console.
 
-### Custom authenticator for Google
+2. Then go to the Service Providers under the main tab. Then add Service Provider name and register it. Let’s use the 
```

Comment:
```suggestion
2. Navigate to **Main** > **Identity** > **Service Providers** > **Add**. 

3. Add a Service Provider name and optionally, add a description. 

4. Click **Register**.
```

Reply:
Fixed https://github.com/wso2/docs-is/pull/2527/commits/ffc80dd6ac9088892baa7c56e5f03a9aa0768d3f